samples/scanner/scanners/firewall_rules/blacklist.yaml

```diff
@@ -0,0 +1,154 @@
+#   a. Checking if a rule applies to the specific resource.
+#   b. Checking each parent in the hierarchy for a rule that applies to that parent.
+#     See: https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy
+#   c. Return the first applicable rules.
+# 3. For each applicable rule, check if the policy is a superset of one of the "match_policies" defined in the rule.
+# 4. For any rule with a matching policy, check if the policy is a superset of any of the "verify_policies"
+#     if it is, create a RuleViolation.
+rules:
+
+    # This rule will trigger on policies that allow ingress for any protocol or port.
+    #
+    # Example triggering firewall rule:
+    #   {
+    #     "name": "firewall1",
+    #     "network": "https://www.googleapis.com/compute/v1/projects/yourproject/global/networks/default",
+    #     "priority": 1000,
+    #     "sourceRanges": ["0.0.0.0/0"],
+    #     "allowed": [
+    #       {
+    #         "IPProtocol": "all",
+    #       }
+    #     ],
+    #     "direction": "INGRESS",
+    #   }
+    #
+    # Example firewall rule that WON'T trigger:
+    #   {
+    #     "name": "firewall2",
+    #     "network": "https://www.googleapis.com/compute/v1/projects/yourproject/global/networks/default",
+    #     "priority": 1000,
+    #     "sourceRanges": ["0.0.0.0/0"],
+    #     "allowed": [
+    #       {
+    #         "IPProtocol": "all",
+    #       }
+    #     ],
+    #     "direction": "EGRESS",
+    #   }
+  - rule_id: 'prevent_allow_all'
+    description: Prevent allow all policies
+    type: blacklist
+    match_policies:
+      - direction: ingress
+        allowed: ['*']
+    verify_policies:
+      - allowed:
+        - IPProtocol: 'all'
+
+    # This rule will trigger on policies that expose all ports for either TCP or UDP.
+    #
+    # Example triggering firewall rule:
+    #   {
+    #     "name": "firewall1",
+    #     "network": "https://www.googleapis.com/compute/v1/projects/yourproject/global/networks/default",
+    #     "priority": 1000,
+    #     "sourceRanges": ["10.0.0.0/8"],
+    #     "allowed": [
+    #       {
+    #         "IPProtocol": "tcp",
+    #         "ports": ["all"]
+    #       }
+    #     ],
+    #     "direction": "INGRESS",
+    #   }
+    #
+    # Example firewall rule that WON'T trigger:
+    #   {
+    #     "name": "firewall2",
+    #     "network": "https://www.googleapis.com/compute/v1/projects/yourproject/global/networks/default",
+    #     "priority": 1000,
+    #     "sourceRanges": ["10.0.0.0/8"],
+    #     "denied": [
+    #       {
+    #         "IPProtocol": "tcp",
+    #         "ports": ["all"]
+    #       }
+    #     ],
+    #     "direction": "INGRESS",
+    #   }
+  - rule_id: 'disallow_any_port'
+    description: Don't allow policies that expose every port
+    type: blacklist
+    match_policies:
+      - direction: ingress
+        allowed: ['*']
+    verify_policies:
```

Comment:
question: is the sourceRanges able to be configured here in the rule?

Reply:
Yes it is.